Assets/MRTK/SDK/Features/UX/Scripts/BoundsControl/Visuals/ScaleHandles.cs

```diff
@@ -85,93 +97,115 @@ protected override void RecreateVisuals()
         /// </summary>
+        /// <param name="handleIndex">cornerIndex</param>
         /// <param name="parent">parent of visual</param>
         /// <param name="isFlattened">instantiate in flattened mode - slate</param>
         /// <returns>bounds of the created visual</returns>
-        private Bounds CreateVisual(GameObject parent, bool isFlattened)
+        private Bounds CreateVisual(int handleIndex, GameObject parent, bool isFlattened)
         {
             // figure out which prefab to instantiate
-            GameObject cornerVisual = null;
+            GameObject handleVisual = null;
             areHandlesFlattened = isFlattened;
             GameObject prefabType = isFlattened ? config.HandleSlatePrefab : config.HandlePrefab;
             if (prefabType == null)
             {
                 // instantiate default prefab, a cube with box collider
-                cornerVisual = GameObject.CreatePrimitive(PrimitiveType.Cube);
-                cornerVisual.transform.parent = parent.transform;
-                cornerVisual.transform.localPosition = Vector3.zero;
+                handleVisual = GameObject.CreatePrimitive(PrimitiveType.Cube);
+                
                 // deactivate collider on visuals and register for deletion - actual collider
                 // of handle is attached to the handle gameobject, not the visual
-                var collider = cornerVisual.GetComponent<BoxCollider>();
+                var collider = handleVisual.GetComponent<BoxCollider>();
                 collider.enabled = false;
-                Object.Destroy(collider);
+                UnityEngine.Object.Destroy(collider);
             }
             else
             {
-                cornerVisual = GameObject.Instantiate(prefabType, parent.transform);
+                handleVisual = GameObject.Instantiate(prefabType, parent.transform);
             }
+            
+            // this is the size of the corner visuals
+            var handleVisualBounds = VisualUtils.GetMaxBounds(handleVisual);
+            float maxDim = VisualUtils.GetMaxComponent(handleVisualBounds.size);
+            float invScale = maxDim == 0.0f ? 0.0f : config.HandleSize / maxDim;
+
+            handleVisual.name = visualsName;
+            handleVisual.transform.parent = parent.transform;
+            handleVisual.transform.localScale = new Vector3(invScale, invScale, invScale);
+            handleVisual.transform.localPosition = Vector3.zero;
+            handleVisual.transform.localRotation = Quaternion.identity;
 
             if (isFlattened)
             {
                 // Rotate 2D slate handle asset for proper orientation
-                cornerVisual.transform.Rotate(0, 0, -90);
+                parent.transform.Rotate(0, 0, -90);
+            } else {
+                Quaternion realignment = GetRotationRealignment(handleIndex);
+                parent.transform.localRotation = realignment;
+            }
```

Comment:
nit on the formatting for consistency:

```suggestion
            }
            else
            {
                Quaternion realignment = GetRotationRealignment(handleIndex);
                parent.transform.localRotation = realignment;
            }
```

Reply:
Do we really not do same-line elses for readability? I'll accept this suggestion, but IMO same-line elses are a god-send for readability and compactness.

Reply:
Happy for this to be something we revisit with the coding guidelines! But yeah, so far, we've kept with the full multi-line brace syntax:

https://docs.microsoft.com/windows/mixed-reality/mrtk-unity/contributing/coding-guidelines?view=mrtkunity-2021-05#use-braces

Largely inspired by https://docs.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions#layout-conventions and the samples formatting.